File: java/problem-1/Problem1.java
public class Problem1 {
    public static void main(String[] args) {
        final int LIMIT = 1000;

        int sumThree = sumMultiples(3, LIMIT);
        int sumFive = sumMultiples(5, LIMIT);
        int sumCommon = sumMultiples(15, LIMIT);

        int answer = sumThree + sumFive - sumCommon;

        System.out.println(answer);
    }

    private static int sumMultiples(int base, int limit) {
        int sum = 0;
        for (int i = base; i < limit; i += base) {
            sum += i;
        }
        return sum;
    }
}

File: go/problem-1/problem-1.go
package main

import "fmt"

func main() {
    const limit = 999

    sumThree := sumMultiples(3, limit)
    sumFive := sumMultiples(5, limit)
    sumCommon := sumMultiples(15, limit)

    answer := sumThree + sumFive - sumCommon

    fmt.Println(answer)
}

// http://www.algebra.com/algebra/homework/Sequences-and-series.faq.question.427328.html
func sumMultiples(multiple int, limit int) (sum int) {
    numTerms := limit / multiple
    lastTerm := multiple * numTerms
    sum = int((float32(numTerms) / 2.0) * float32(multiple + lastTerm))
    return
}
